src/ui/hooks/tests.ts

```diff
@@ -230,7 +166,17 @@ export function useGetTestRunsForWorkspace(workspaceId: WorkspaceId): {
     }
     assert("testRuns" in data.node, "No testRuns in GetTestsRun response");
 
-    const testRuns = data.node.testRuns?.map(testRun => convertTestRun(testRun)!);
+    const testRuns: TestRun[] = [];
+
+    // Convert legacy test runs; filter out ones with invalid data
+    data.node.testRuns?.forEach(legacyTestRun => {
+      try {
+        testRuns.push(convertLegacyTestRunToNewFormat(legacyTestRun));
+      } catch (error) {
+        // Filter out and ignore data that's too old or corrupt to defined the required fields
+      }
```

Comment:
A lot of records seem to be missing required data. Ryan mentioned:
> If the replay was uploaded from local dev or from an unsupported CI platform, those fields can be empty

I think for now the best thing we can do is just filter them out.